Java/core/src/main/java/com/amazon/randomcutforest/state/tree/CompactRandomCutTreeFloatMapper.java

```diff
@@ -40,9 +42,16 @@ public CompactRandomCutTreeFloat toModel(CompactRandomCutTreeState state, Compac
 
         INodeStore nodeStore;
 
-        NodeStoreMapper nodeStoreMapper = new NodeStoreMapper();
-        nodeStoreMapper.setPartialTreeStateEnabled(state.isPartialTreeState());
-        nodeStore = nodeStoreMapper.toModel(state.getNodeStoreState());
+        if (state.getMaxSize() < SmallNodeStore.MAX_SMALLNODESTORE_CAPACITY
```

Comment:
do we have functional tests covering both branches using SmallNodeStore and NodeStore?

Reply:
Good callout. Adding tests for SmallNodeStore. Let me add functional tests for both branches -- we have not tested with sample size > 16k at too many places, but there is no reason it cannot be used with large sample sizes.